homeassistant/components/livisi/shc.py

```diff
@@ -171,32 +82,54 @@ async def async_set_all_rooms(self):
         response = await self.aiolivisi.async_get_all_rooms()
 
         for available_room in response:
-            self._rooms[available_room.get("id")] = available_room.get("config").get(
+            self.rooms[available_room.get("id")] = available_room.get("config").get(
                 "name"
             )
 
-    def on_data(self, event_data: str) -> None:
-        """Define a handler to fire when the data is received."""
-        json_data = json.loads(event_data)
-        for device in self.included_devices:
-            source = json_data.get("source")
-            if device.capability_id == source:
-                device.update_states(json_data.get("properties"))
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Set LIVISI devices."""
+        shc_devices: dict[Any, Any] = self.coordinator.data
+        for device in shc_devices:
+            if device in self.devices:
                 continue
-            if device.unique_id in source:
-                reachability = json_data.get("properties")
-                is_reachable: bool = reachability.get("isReachable")
-                device.update_reachability(is_reachable)
-            elif device.available is False:
-                device.update_reachability(True)
+            if device["type"] == PSS_DEVICE_TYPE:
+                if device not in self.switch_devices:
+                    self.switch_devices.append(device)
+                    dispatcher_send(self._hass, LIVISI_DISCOVERY_NEW, device)
+            self.devices.append(device)
+
+    def on_data(self, event_data: LivisiEvent) -> None:
+        """Define a handler to fire when the data is received."""
+        source = event_data.source
+        if event_data.onState is not None:
+            device_id_state: dict = {
+                "id": source,
+                "state": event_data.onState,
+            }
+            dispatcher_send(self._hass, LIVISI_STATE_CHANGE, device_id_state)
```

Comment:
To notify the coordinator of new pushed data, use `DataUpdateCoordinator.async_set_updated_data`.